### PR TITLE
[JENKINS-30224] use real listener to prevent NPE

### PR DESCRIPTION
**Pull request by ydubreuil**
@reviewbybees

### CODE REVIEW

File: src/main/java/org/jvnet/hudson/tools/versionnumber/VersionNumberBuilder.java [hunk 1]
```diff
@@ -494,7 +495,7 @@ private static String sizeTo(String s, int length) {
     public Environment setUp(AbstractBuild build, Launcher launcher, BuildListener listener) {
         String formattedVersionNumber = "";
         try {
-            VersionNumberBuildInfo info = incBuild(build, build.getEnvironment(listener), listener.getLogger());
+            VersionNumberBuildInfo info = incBuild(build, listener, listener.getLogger());
```
Comment (oleg-nenashev) [inline src/main/java/org/jvnet/hudson/tools/versionnumber/VersionNumberBuilder.java]: Seems you also don't need to specify the logger as a parameter now

**Comment by varmenise** [comment]
looks good to me :bee:

File: src/main/java/org/jvnet/hudson/tools/versionnumber/VersionNumberBuilder.java (hunk 1 above)
Comment (varmenise) [inline src/main/java/org/jvnet/hudson/tools/versionnumber/VersionNumberBuilder.java]: agree, `incBuild` is `private` so the signature can be changed safely

Comment (ydubreuil) [inline src/main/java/org/jvnet/hudson/tools/versionnumber/VersionNumberBuilder.java]: good catch, forgot to convert third argument too

Comment (ydubreuil) [inline src/main/java/org/jvnet/hudson/tools/versionnumber/VersionNumberBuilder.java]: funny, it's not even used...

**Comment by oleg-nenashev** [comment]
:bee:

**Comment by ydubreuil** [comment]
@reviewbybees done

**Pull request merged**